### PR TITLE
Reuse `Channel` type in `direct-fund` protocol

### PR DESCRIPTION
**Pull request by geoknee**
Towards #78. This is a refactor aimed at removing code and ensuring the direct fund protocol does not replicate state that is owned by `Channel`. 

~I will leave this as a draft until I have cherry-picked some of the work into smaller PRs.~

### CODE REVIEW

**Comment by geoknee** [comment]
Note: 16ea777 is superseded by #127. I will drop that commit / rewrite history post review but pre merge.

File: protocols/direct-fund/direct-fund.go [hunk 1]
```diff
@@ -29,56 +29,40 @@ var ErrNotApproved = errors.New("objective not approved")
...
-			init.myIndex = uint(i)
+			myIndex = uint(i)
 		}
 	}
 
+	init.C = &channel.Channel{}
+	*init.C, err = channel.New(initialState, isTwoPartyLedger, myIndex, myDestination, theirDestination)
+
+	if err != nil {
+		return init, err
```
Comment (NiloCK) [inline protocols/direct-fund/direct-fund.go:63]:
```suggestion
		return DirectFundObjective{}, fmt.Errorf("failed to initialize channel for direct-fund objective: %w", err)
```
- I'm preferring explicitly empty object returns in failure cases
- per a discussion w/ @andrewgordstewart yesterday, we should probably make a habit of wrapping errors from called-code in order to improve debuggability

File: protocols/direct-fund/direct-fund_test.go [hunk 2]
```diff
@@ -52,25 +72,25 @@ func TestNew(t *testing.T) {
...
 }
 var dummyState = state.State{}
-var stateToSign state.State = s.expectedStates[0]
-var privateKeyOfParticipant0 = common.Hex2Bytes(`caab404f975b4620747174a75f08d98b4e5a7053b691b41bcfc0d839d48b7634`)
-var correctSignatureByParticipant, _ = stateToSign.Sign(privateKeyOfParticipant0)
+var stateToSign state.State = s.C.PreFundState()
+var correctSignatureByParticipant, _ = stateToSign.Sign(alice.privateKey)
 
 func TestUpdate(t *testing.T) {
+	// fmt.Println(testState)
```
Comment (NiloCK) [inline protocols/direct-fund/direct-fund_test.go:94]:
```suggestion
```

File: protocols/direct-fund/direct-fund_test.go [hunk 3]
```diff
@@ -149,11 +176,11 @@ func TestCrank(t *testing.T) {
 	}
 
 	// Manually progress the extended state by collecting prefund signatures
-	o.(DirectFundObjective).preFundSigned[0] = true
-	o.(DirectFundObjective).preFundSigned[1] = true
+	o.(DirectFundObjective).C.AddSignedState(o.(DirectFundObjective).C.PreFundState(), correctSignatureByAliceOnPreFund)
+	o.(DirectFundObjective).C.AddSignedState(o.(DirectFundObjective).C.PreFundState(), correctSignatureByBobOnPreFund)
```
Comment (NiloCK) [inline protocols/direct-fund/direct-fund_test.go:180]: This is gnarly, and I have a neat way to tidy it, but it doesn't fit nicely as a suggestion (touches a bunch of lines). Will punt to a small follow-up PR.

**Review by NiloCK — APPROVED** [review]
Great.

**Pull request merged**